source/dem/disable_particle_contact.cc

```diff
@@ -0,0 +1,334 @@
+              // temperature
+              for (typename Particles::ParticleHandler<
+                     dim>::particle_iterator_range::iterator
+                     particles_in_cell_iterator = particles_in_cell.begin();
+                   particles_in_cell_iterator != particles_in_cell.end();
+                   ++particles_in_cell_iterator)
+                {
+                  auto &particle_properties =
+                    particles_in_cell_iterator->get_properties();
+
+                  for (int d = 0; d < dim; ++d)
+                    {
+                      cell_velocity_fluctuation_squared_sum[d] +=
+                        (particle_properties[DEM::PropertiesIndex::v_x + d] -
+                         velocity_cell_average[d]) *
+                        (particle_properties[DEM::PropertiesIndex::v_x + d] -
+                         velocity_cell_average[d]);
+                    }
+                }
+
+              // Calculate average granular temperature in the cell
+              for (int d = 0; d < dim; ++d)
+                {
+                  cell_velocity_fluctuation_squared_average[d] =
+                    cell_velocity_fluctuation_squared_sum[d] /
+                    n_particles_in_cell;
+                  granular_temperature_cell +=
+                    (1.0 / dim) * cell_velocity_fluctuation_squared_average[d];
+                }
+            }
+
+          // Store the average granular temperature and solid fraction with
+          // active cell index
+          granular_temperature_average[cell->active_cell_index()] =
+            granular_temperature_cell;
+          solid_fractions[cell->active_cell_index()] = solid_fraction;
+        }
+    }
+}
+
+
+template <int dim>
+void
+DisableParticleContact<dim>::identify_mobility_status(
+  const DoFHandler<dim> &                background_dh,
+  const Particles::ParticleHandler<dim> &particle_handler,
+  MPI_Comm                               mpi_communicator)
+{
+  // Reset cell status containers
+  status_to_cell.clear();
+  status_to_cell.resize(mobility_status::n_mobility_status);
+
+  // Create dummy dofs for background dof handler
+  const FE_Q<dim>    fe(1);
+  const unsigned int dofs_per_cell = fe.dofs_per_cell;
+
+  // Get locally owned and relevant dofs
+  const IndexSet locally_owned_dofs = background_dh.locally_owned_dofs();
+  IndexSet       locally_relevant_dofs =
+    DoFTools::extract_locally_relevant_dofs(background_dh);
+
+  // Reinit all value of mobility at nodes
+  mobility_at_nodes.reinit(locally_owned_dofs,
+                           locally_relevant_dofs,
+                           mpi_communicator);
+  mobility_at_nodes = 0;
+
+  // Empty status (3) to nodes when no particle in cell
+  for (auto cell = active_ghost_cells.begin();
+       cell != active_ghost_cells.end();)
+    {
+      // Check if the cell has any particles
+      if (particle_handler.n_particles_in_cell(*cell) == 0)
```

Comment:
Could you pass by reference instead of using a pointer here to follow the pattern?

Reply:
I think it didn't work and that's why I used a pointer. Maybe I should do something at line 47 So iteration works.

Reply:
I think you just need to use reference in line 172 instead

Reply:
Yeah, it doesn't work, I think I tried to do that.
Actually, it could work with `(auto &cell : active_ghost_cells)`, but this iterator doesn't work with .erase().
I will keep this comment open, so I could look at that later

Reply:
Ok! Did you try dereferencing .begin and .end?

Reply:
Just tried it, but still won't work with erase(), cell needs to be the iterator